projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js

```diff
@@ -48,55 +42,47 @@ const PublicizePanel = ( { prePublish } ) => {
 		[ hasEnabledConnections, refresh ]
 	);
 
-	// Disable the panel when no proper site plan is available.
-	const PanelRowWithDisabled = isPublicizeDisabledBySitePlan ? Disabled : PanelRow;
-
 	// Panel wrapper.
 	const PanelWrapper = prePublish ? Fragment : PanelBody;
 	const wrapperProps = prePublish
 		? {}
 		: {
 				title: __( 'Share this post', 'jetpack' ),
-				className: isPublicizeDisabledBySitePlan ? 'jetpack-publicize-disabled' : '',
 		  };
 
 	return (
 		<PanelWrapper { ...wrapperProps }>
-			<UpsellNotice isPostPublished={ isPostPublished } />
-
-			{ ! hideRePublicizeFeature && (
-				<Fragment>
-					{ isRePublicizeFeatureEnabled && ! isPostPublished && (
-						<PanelRowWithDisabled>
-							<ToggleControl
-								className="jetpack-publicize-toggle"
-								label={
-									isPublicizeEnabled && ! isPublicizeDisabledBySitePlan
-										? __( 'Share when publishing', 'jetpack' )
-										: __(
-												'Sharing is disabled',
-												'jetpack',
-												/* dummy arg to avoid bad minification */ 0
-										  )
-								}
-								onChange={ togglePublicizeFeature }
-								checked={ isPublicizeEnabled }
-								disabled={ ! hasConnections }
-							/>
-						</PanelRowWithDisabled>
-					) }
+			<Fragment>
+				{ ! isPostPublished && (
+					<PanelRow>
+						<ToggleControl
+							className="jetpack-publicize-toggle"
+							label={
+								isPublicizeEnabled
+									? __( 'Share when publishing', 'jetpack' )
+									: __(
+											'Sharing is disabled',
+											'jetpack',
+											/* dummy arg to avoid bad minification */ 0
+									  )
+							}
+							onChange={ togglePublicizeFeature }
+							checked={ isPublicizeEnabled }
+							disabled={ ! hasConnections }
+						/>
+					</PanelRow>
+				) }
 
-					<PublicizeConnectionVerify />
-					<PublicizeForm
-						isPublicizeEnabled={ isPublicizeEnabled }
-						isRePublicizeFeatureEnabled={ isRePublicizeFeatureEnabled }
-						isPublicizeDisabledBySitePlan={ isPublicizeDisabledBySitePlan }
-					/>
-					{ isPublicizeEnabled && <PublicizeTwitterOptions prePublish={ prePublish } /> }
+				<PublicizeConnectionVerify />
+				<PublicizeForm
+					isPublicizeEnabled={ isPublicizeEnabled }
+					isRePublicizeFeatureEnabled={ true }
+					isPublicizeDisabledBySitePlan={ false }
```

Comment:
We can probably get rid of these altogether and clean up the `PublicizeForm` component, but we can do that in a separate PR.

Reply:
Sure thing!